### ISSUE TITLE
ES7 decorator: TypeError: Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by DanSnow** [eslint/eslint#5895]
Hello, I'm using eslint v2.8.0, with babel-eslint.

My configure:
```json
{
  "parser"  : "babel-eslint",
  "extends" : [
    "standard",
    "standard-react"
  ],
  "env"     : {
    "browser" : true
  },
  "globals" : {
    "__DEV__"      : false,
    "__PROD__"     : false,
    "__DEBUG__"    : false,
    "__DEBUG_NEW_WINDOW__" : false,
    "__BASENAME__" : false
  },
  "rules": {
    "semi" : [2, "always"],
    "space-before-function-paren" : [2, "never"],
    "react/jsx-closing-bracket-location": [2, "after-props"],
    "arrow-parens": [0, "as-needed"],
    "react/jsx-curly-spacing": [2, "always"],
    "quotes": [2, "single", "avoid-escape"]
  }
}
```

Here is the code:
```js
class Test {
  @autobind
  foo() {
  }
}
```

It should report decorator "autobind" undefined, but it crash.

```
Cannot read property 'type' of undefined
TypeError: Cannot read property 'type' of undefined
    at hasTypeOfOperator (node_modules/eslint/lib/rules/no-undef.js:21:18)
    at /home/snow/Desktop/Juice/node_modules/eslint/lib/rules/no-undef.js:39:40
    at Array.forEach (native)
    at EventEmitter.Program:exit (node_modules/eslint/lib/rules/no-undef.js:36:33)
    at emitOne (events.js:95:20)
    at EventEmitter.emit (events.js:182:7)
    at NodeEventGenerator.leaveNode (node_modules/eslint/lib/util/node-event-generator.js:51:22)
    at CodePathAnalyzer.leaveNode (node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:629:23)
    at CommentEventGenerator.leaveNode (node_modules/eslint/lib/util/comment-event-generator.js:112:23)
    at Controller.traverser.traverse.leave (node_modules/eslint/lib/eslint.js:890:36)

```

**Comment by ilyavolodin** [comment]
You should open an issue about this with [babel-eslint](https://github.com/babel/babel-eslint) instead, since we do not support experimental features, and none of our rules are written to take into account things like decorators.
Closing, as this is not something we support.
